consensus/src/transaction_shuffler/mod.rs

```diff
@@ -19,6 +25,20 @@ pub mod transaction_shuffler_fuzzing {
 /// Interface to shuffle transactions
 pub trait TransactionShuffler: Send + Sync {
     fn shuffle(&self, txns: Vec<SignedTransaction>) -> Vec<SignedTransaction>;
+
+    /// Given a configuration and a vector of SignedTransactions, return an iterator that
+    /// produces them in a particular shuffled order.
+    fn signed_transaction_iterator(
+        &self,
+        txns: Vec<SignedTransaction>,
+    ) -> Box<dyn Iterator<Item = SignedTransaction> + 'static>;
+
+    /// Given a configuration and a vector of SignatureVerifiedTransaction, return an iterator of
+    /// SignedTransaction.
```

Comment:
nit: The iterator returned is also SignatureVerifiedTransaction

Reply:
Fixed, good catch